temporal-kotlin/src/main/kotlin/io/temporal/internal/async/KotlinMethodReferenceDisassemblyService.kt

```diff
@@ -22,24 +22,34 @@ package io.temporal.internal.async
 import io.temporal.internal.async.spi.MethodReferenceDisassemblyService
 import io.temporal.workflow.Functions
 import kotlin.jvm.internal.CallableReference
+import kotlin.jvm.internal.Lambda
 
 class KotlinMethodReferenceDisassemblyService : MethodReferenceDisassemblyService {
   override fun getMethodReferenceTarget(methodReference: Any): Any? {
-    if (methodReference is CallableReference) {
-      return unwrapCallableReference(methodReference)
+    return when (methodReference) {
+      is Functions.TemporalFunctionalInterfaceMarker -> {
+        unwrapTemporalFunctionalInterfaceInKotlin(methodReference)
+      }
+      else -> {
+        unwrapIfCallableReference(methodReference)
+      }
     }
+  }
 
-    if (methodReference is Functions.TemporalFunctionalInterfaceMarker) {
-      return unwrapTemporalFunctionalInterfaceInKotlin(methodReference)
+  private fun unwrapIfCallableReference(callableReference: Any): Any? {
+    return when (callableReference) {
+      /**
+       * Strategy 1.1
+       * We unwrap simple native Kotlin [CallableReference] (which is used for method references)
+       */
+      is CallableReference -> unwrapCallableReference(callableReference)
+      // we end up here if lambda is passed instead of a method reference
+      is Lambda<*> -> null
+      // something unexpected that we don't know how to handle
+      else -> null
```

Comment:
Higher up does this result in us throwing an error? I'm wondering if maybe we can provide a little additional context or info down here somehow. Not a big deal either way.

Reply:
Nah, it will result in our inability to extract a target and we will schedule it as a separate thread just as an async function, not temporal stub method reference.